api/maven-api-core/src/main/java/org/apache/maven/api/Type.java

```diff
@@ -80,11 +129,16 @@ public interface Type {
     String getClassifier();
 
     /**
-     * Specifies if the artifact contains java classes and should be
-     * added to the classpath.
+     * Specifies if the artifact contains java classes and can be added to the classpath.
+     * Whether the artifact <em>should</em> be added to the classpath depends on other
+     * {@linkplain #getDependencyProperties() dependency properties}.
+     *
+     * @return if the artifact <em>can</em> be added to the class path
      *
-     * @return if the artifact should be added to the class path
+     * @deprecated A value of {@code true} does not mean that the dependency <em>should</em>
+     * be placed on the classpath. See {@link JavaPathType} instead for better analysis.
      */
+    @Deprecated
```

Comment:
If deprecation is necessary, it should be noted that it will be removed before 4.0.0 GA.